src/webview/VSCodeKaotoEditorChannelApi.ts

```diff
@@ -1,8 +1,20 @@
+import { KaotoEditorChannelApi } from '@kaoto/kaoto';
+import { ISettingsModel, SettingsModel } from '@kaoto/kaoto/models';
 import { DefaultVsCodeKieEditorChannelApiImpl } from '@kie-tools-core/vscode-extension/dist/DefaultVsCodeKieEditorChannelApiImpl';
 import * as vscode from 'vscode';
 
-export class VSCodeKaotoEditorChannelApi extends DefaultVsCodeKieEditorChannelApiImpl {
-    async getCatalogURL(): Promise<string | undefined> {
-        return await vscode.workspace.getConfiguration('kaoto').get('catalog.url');
-    }
+export class VSCodeKaotoEditorChannelApi
+  extends DefaultVsCodeKieEditorChannelApiImpl
+  implements KaotoEditorChannelApi
+{
+  async getCatalogURL(): Promise<string | undefined> {
+    return await vscode.workspace.getConfiguration('kaoto').get('catalog.url');
+  }
+
+  async getVSCodeKaotoSettings(): Promise<ISettingsModel> {
+    const catalogUrl =
+      vscode.workspace.getConfiguration('kaoto').get<string | null>('catalog.url') ?? '';
```

Comment:
if I understand well, we now return an empty string in case the setting is not set (so user let to its default)?
Is it fine to no more distinguish between the 2 states? (I guess that in the case of catalogUrl it is fine)

Reply:
Yes, so far, the `catalogUrl` from VSCode can be `null` if the user hasn't set it or a `string` that could be empty or not, but on the Kaoto side, `catalogUrl` can only be a string (no null) hence we return a string here.